src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -9,9 +9,264 @@
 #include "bscu.h"
 #include "slurm.h"
 
+#if defined(HAVE_SLURM_SLURM_H)
+#include <slurm/slurm.h> /* for slurm_hostlist_create */
+#elif defined(HAVE_POSIX_REGCOMP)
+#include <regex.h> /* for POSIX regular expressions */
+
+#define MAX_GMATCH 5 /* max number of atoms in group matches + 1 */
+#define MAX_RMATCH 5 /* max number of atoms in range matches + 1 */
+#define MAX_EMATCH 2 /* max number of atoms in element matches + 1 */
+#define MAX_NNODES_STRLEN 8 /* max string len for node # incl. '\0' */
```

Comment:
What does this limit? The comment makes me concerned that hostnames > 7 characters would be problematic.

Reply:
cc @gcongiu

Reply:
@raffenet that is the numerical part of the hostname. I am limiting it to 10 millions but we can extend it if this may become a concern.

Reply:
Ah, got it.